### ISSUE TITLE
[BUG] objects are being documented while they are not present in correct location or absent altogether

### ISSUE THREAD

**Issue by yarnabrina** [sktime/sktime#4803]
Currently, `make docs`lead to the following warnings during build process:

```shell
WARNING: [autosummary] failed to import 'sktime.classification.deep_learning.MACNNClassifier': no module named sktime.classification.deep_learning.MACNNClassifier
WARNING: [autosummary] failed to import 'sktime.performance_metrics.forecasting.count_error': no module named sktime.performance_metrics.forecasting.count_error
WARNING: [autosummary] failed to import 'sktime.performance_metrics.forecasting.hausdorff_error': no module named sktime.performance_metrics.forecasting.hausdorff_error
WARNING: [autosummary] failed to import 'sktime.performance_metrics.forecasting.prediction_ratio': no module named sktime.performance_metrics.forecasting.prediction_ratio
WARNING: [autosummary] failed to import 'sktime.regression.deep_learning.ResNetRegressor': no module named sktime.regression.deep_learning.ResNetRegressor
WARNING: [autosummary] failed to import 'sktime.transformations.panel.compose.SeriesToPrimitivesRowTransformer': no module named sktime.transformations.panel.compose.SeriesToPrimitivesRowTransformer
WARNING: [autosummary] failed to import 'sktime.transformations.panel.compose.SeriesToSeriesRowTransformer': no module named sktime.transformations.panel.compose.SeriesToSeriesRowTransformer
WARNING: [autosummary] failed to import 'sktime.transformations.panel.compose.make_row_transformer': no module named sktime.transformations.panel.compose.make_row_transformer
```

I have checked for each of these, and most of these are valid issues:

1. `SeriesToPrimitivesRowTransformer`, `SeriesToSeriesRowTransformer`, `make_row_transformer` are absent in codebase but are being listed in `docs/source/api_reference/transformations.rst`. The first two are also present in `sktime/tests/_config.py`
2. `count_error`, `hausdorff_error`, `prediction_ratio` are present in `annotation` but are being listed in `docs/source/api_reference/performance_metrics.rst` as part of `forecasting`
3. `MACNNClassifier` is defined in `sktime/classification/deep_learning/macnn.py`, but not imported and exposed through `sktime/classification/deep_learning/__init__.py` file, even though listed as such in `docs/source/api_reference/classification.rst`
4. `ResNetRegressor` is defined in `sktime/regression/deep_learning/resnet.py`, but not imported and exposed through `sktime/regression/deep_learning/__init__.py` file, even though listed as such in `docs/source/api_reference/regression.rst`

**Comment by fkiraly** [comment]
1. the `RowTransformer`-s have been removed a while ago as they were adding the row broadcasting capability that all transformerws have by default. This should be removed from the docs too.
2. look like a typo
3. , 4. look like omissions, the estimators should be exposed in the module `__init__`.

**Comment by fkiraly** [comment]
`performance_metrics` should be addressed by https://github.com/sktime/sktime/pull/4813
row transformers should be addressed by https://github.com/sktime/sktime/pull/4895

this leaves the two neural networks then?

**Comment by yarnabrina** [comment]
Yes (till 2 weeks back, hopefully nothing new came up since).

**Comment by fkiraly** [comment]
Can we upgrade this to a test failure once fixed?

**Comment by fkiraly** [comment]
btw, this bug report https://github.com/sktime/sktime/issues/4905 has the opposite problem - estimators that exist but are not in the API overview.

The estimator overview table hints at how one could test for that!

**Comment by yarnabrina** [comment]
> Can we upgrade severity of this to a test failure once fixed?

I think it's possible from `pre-commit` hooks. Here's the reference:

https://numpydoc.readthedocs.io/en/latest/validation.html

Right now, read the docs logs show lots of warnings, and I remember getting a lot while building docs locally. If we enable this, that'd be a very tedious PR to handle all issues I think.

**Comment by fkiraly** [comment]
@yarnabrina, I looked through the list of the `numpydocs` tests (which can be turned on individually), but couldn't find sth that relates to broken links. Can you quickly check whether this is the link you wanted ot post? If yes, which exact test number(s) would do what we want?

**Comment by fkiraly** [comment]
> not just for broken links

Yes, but what I mean, the `numpydoc` tests don't seem to include a test for broken links?

**Comment by yarnabrina** [comment]
Actually, I'm not much surprised. `numpydoc`'s whole idea is style of docstrings, and creating a sort of standard around it. This really does not fall into that.

(That also means I shouldn't have suggested `numpydoc` in the first place, but I misinterpreted earlier that you're planning to solve all the warnings that Read the Docs logs show currently.)

**Comment by fkiraly** [comment]
> misinterpreted earlier that you're planning to solve all the warnings that Read the Docs logs show currently.

Ahhhh - that's also a good idea! Should we track that, if we don't already?
